### ISSUE TITLE
add total supply to token storage

### ISSUE THREAD

**Issue by tzemanovic** [anoma/namada#338]
We should add the total supply to the token VP to avoid iteration (the VP can simply reject any changes to it in a transaction as only the protocol can mint tokens). This is needed for inflation system, see https://github.com/anoma/namada/pull/230#pullrequestreview-1056990194

**Comment by cwgoes** [comment]
@brentstone perhaps this is covered as part of the proof-of-stake inflation work?

**Comment by brentstone** [comment]
> @brentstone perhaps this is covered as part of the proof-of-stake inflation work?

Yes, didn't notice there was an explicit issue for this. It is in #714.
